### PR TITLE
feat: add FeeTapper

### PR DESCRIPTION
**Pull request by zhongeric**
The recipient is intended to be `FeeTapper`, a singleton contract which holds a multitude of active deposits. Each currency has a unique `Tap`, which tracks its balance over time. Each call to `sync` creates a new deposit, also called `Keg`. Each `Keg` holds a per block release amount and an end block. Kegs can be released in batch (all at once per currency, or one by one). Kegs are stored in a flat linked list in storage and Taps hold pointers to the head/tail within the list.

`FeeTapper` streams fees over time, ensuring that large fee payments are incrementally sent to the fee recipient. For more details on the token jar implementation, check out the https://github.com/Uniswap/protocol-fees/blob/main/src/releasers/ExchangeReleaser.sol  

---
## Summary
Adds the FeeTapper contract, a singleton that handles streaming protocol fees to TokenJar using a configurable per-block release rate to smooth fee distribution.
## Changes
- Add `src/feeAdapters/FeeTapper.sol` - Singleton contract managing fee streaming with linked-list "kegs" for each currency "tap"
- Add `src/interfaces/IFeeTapper.sol` - Interface defining Tap/Keg structs and contract methods
- Add `test/FeeTapper.t.sol` - Comprehensive test suite (573 lines) covering sync, release, and edge cases
- Add `snapshots/FeeTapperTest.json` - Gas benchmarks for operations
## Key Features
- **Linear fee release**: Fees are released over time based on `perBlockReleaseRate` (default 10 bps/block = 1000 blocks for full release)
- **Multi-currency support**: Handles both native ETH and ERC20 tokens
- **Keg management**: Each deposit creates a "keg" with its own release schedule; empty kegs are automatically cleaned up via linked list operations
- **Owner controls**: Release rate configurable by owner (must evenly divide 10000 bps)
- **Gas efficient**: Single keg release ~70k gas, multiple keg release with cleanup ~88k gas

### CODE REVIEW

File: src/feeAdapters/FeeTapper.sol [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+/// @notice Singleton contract which handles the streaming of incoming protocol fees to TokenJar
+contract FeeTapper is IFeeTapper, Ownable {
+  using CurrencyLibrary for Currency;
+
+  address public immutable tokenJar;
+
+  /// @notice Mapping of currencies to Taps
+  mapping(Currency => Tap) private $_taps;
+  /// @notice Linked list of kegs. Taps manage the head and tail of their respective kegs.
+  mapping(uint32 => Keg) private $_kegs;
```
Comment (hensha256) [inline src/feeAdapters/FeeTapper.sol]: At a high level - whats the motivation to have multiple kegs per currency? If the contract has say $1m USDC in it, dont we want to limit the total USDC that can get dripped in each block? That to me feels like 1 tap that just gets topped up over time with each deposit and no separation of different release schedules for different sets of tokens. But maybe theres a specific reason for that?
